pkg/operator/baremetal_pod.go

```diff
@@ -77,33 +128,95 @@ func setMariadbPassword() corev1.EnvVar {
 	}
 }
 
-func generateRandomPassword() string {
-	rand.Seed(time.Now().UnixNano())
+func setIronicHtpasswdHash(name string, secretName string) corev1.EnvVar {
+	return corev1.EnvVar{
+		Name: name,
+		ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: secretName,
+				},
+				Key: ironicHtpasswdKey,
+			},
+		},
+	}
+}
+
+func generateRandomPassword() (string, error) {
 	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
 		"abcdefghijklmnopqrstuvwxyz" +
 		"0123456789")
 	length := 16
 	buf := make([]rune, length)
+	numChars := big.NewInt(int64(len(chars)))
 	for i := range buf {
-		buf[i] = chars[rand.Intn(len(chars))]
+		c, err := rand.Int(rand.Reader, numChars)
+		if err != nil {
+			return "", err
+		}
+		buf[i] = chars[c.Uint64()]
 	}
-	return (string(buf))
+	return string(buf), nil
 }
 
 func createMariadbPasswordSecret(client coreclientv1.SecretsGetter, config *OperatorConfig) error {
-	glog.V(3).Info("Checking if the Maridb password secret already exists")
+	glog.V(3).Info("Checking if the MariaDB password secret already exists")
 	_, err := client.Secrets(config.TargetNamespace).Get(context.Background(), baremetalSecretName, metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
+		password, err := generateRandomPassword()
+		if err != nil {
+			return err
+		}
 		// Secret does not already exist. So, create one.
-		_, err := client.Secrets(config.TargetNamespace).Create(
+		_, err = client.Secrets(config.TargetNamespace).Create(
 			context.Background(),
 			&corev1.Secret{
 				ObjectMeta: metav1.ObjectMeta{
 					Name:      baremetalSecretName,
 					Namespace: config.TargetNamespace,
 				},
 				StringData: map[string]string{
-					baremetalSecretKey: generateRandomPassword(),
+					baremetalSecretKey: password,
+				},
+			},
+			metav1.CreateOptions{},
+		)
+		return err
+	}
+	return err
+}
+
+func createIronicPasswordSecret(client coreclientv1.SecretsGetter, config *OperatorConfig, name string, username string, configSection string) error {
+	glog.V(3).Info(fmt.Sprintf("Checking if the %s password secret already exists", name))
+	_, err := client.Secrets(config.TargetNamespace).Get(context.Background(), name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
```

Comment:
Reverse this check and return early with the unexpected error?

Reply:
Yeah, that would have been better style. I just copied and pasted from the existing createMariadbPasswordSecret function. Maybe we can follow up later with a refactor?